File: Week3/assignment.py
# info是捷運站
# stitle是景點
# address區域
# file有圖片
# longitude經度
# latitude緯度
import os
import json
import csv
from urllib.request import urlopen
# 讀取
url = "https://padax.github.io/taipei-day-trip-resources/taipei-attractions-assignment.json"
response = urlopen(url)
data = json.load(response)

fields = ['stitle', 'address', 'longitude', 'latitude','file']
output_file = 'Week3/tour.csv'
if not os.path.exists(output_file):
    with open(output_file, 'w', newline='') as file:
        writer = csv.DictWriter(file, fieldnames=fields)

        # 寫入CSV 
        writer.writeheader()

        # 寫入每一筆資料
        for item in data['result']['results']:
            address = item['address'].split('市', 1)[1].split('區', 1)[0]+ "區"
            capital = item['file'].split('JPG',1)[0]+'jpg'
            lower = item['file'].split('jpg',1)[0]+'jpg'
            if len(capital)<len(lower):
                image_url = item['file'].split('JPG',1)[0]+'jpg'
            else:
                image_url = item['file'].split('jpg',1)[0]+'jpg'
            row = {
                'stitle': item['stitle'],
                'address': address,
                'longitude': item['longitude'],
                'latitude': item['latitude'],
                'file': image_url
            }
            writer.writerow(row)
    print("finished")
else:
    print("tour existed")
# list=[]
# for i in data['result']['results']:
#     if i["MRT"] in list:
#         continue
#     else:
#         list.append(i["MRT"])
# print(list)
# print(len(list))
mrt_dic={}
for i in data['result']['results']:
    mrt = i['MRT']
    if mrt not in mrt_dic:
        mrt_dic[mrt]=[]
    mrt_dic[mrt].append(i['stitle'])
output_file2="Week3/mrt_tour.csv"

if not os.path.exists(output_file2):
    with open(output_file2,'w',newline='')as file:
        writer = csv.writer(file)
        fields = ['MRT station'] + [f'Attraction{i+1}' for i in range(max(len(attractions) for attractions in mrt_dic.values()))]
        #表格填寫mrt，後續補上景點，首先先將所有景點遍歷找出所有景點的字典長度(此字典包含mrt:attractions)，
        #然後找出最大的值，再將最大值列入另一迴圈寫入表格(按照1,2,3...)，但因為索引值從一開始，統一加一。
        writer.writerow(fields)
        print(fields)
        for mrt,attractions in mrt_dic.items():
            row_data = [mrt] + attractions#attractions是value的list
            # 缺少欄位補上空值
            if len(row_data) < len(fields):
                row_data.extend([''] * (len(fields) - len(row_data)))
            writer.writerow(row_data)
        print("mrt finished")
else:
    print("mri existed")

File: Week3/assignment.js
console.log("js succeed");
fetch('https://padax.github.io/taipei-day-trip-resources/taipei-attractions-assignment.json')
    .then(response => response.json())
    .then(data => {
        console.log("fetch succeed");
        const attractions = data.result.results; // results是一個array
        let titles = [];
        let images = [];

        // 取得前三個attractions的title和image
        for (let i = 0; i < 3; i++) {
            const title = attractions[i].stitle;
            console.log("for內", title);
            titles.push(title);

            const firstImageUrl = attractions[i].file.split("https://")[1];
            console.log("First Image URL: " + "https://" + firstImageUrl);
            images.push("https://" + firstImageUrl);
        }

        console.log("for外", titles);
        console.log("for外", images);

        // 將前三個attractions的title和image顯示在promotion區塊
        for (let i = 0; i < 3; i++) {
            const promotionDiv = document.getElementById(`promotion${i + 1}`);
            const promotionImage = promotionDiv.querySelector('img');
            const promotionTitle = promotionDiv.querySelector('.title');

            promotionImage.src = images[i];
            promotionTitle.textContent = titles[i];
        }

        // 將後續的區塊插入圖片和文字
        for (let i = 3; i < attractions.length; i++) {
            const title = attractions[i].stitle;
            console.log("for內", title);
            titles.push(title);

            const imageUrl = attractions[i].file.split("https://")[1];
            console.log("Image URL: " + "https://" + imageUrl);
            images.push("https://" + imageUrl);

            const titlePicDiv = document.createElement('div');
            titlePicDiv.classList.add('titlepic');

            const titleImage = document.createElement('img');
            titleImage.src = images[i];
            titleImage.classList.add('image');
            titleImage.id = `pic${i + 1}`;

            const titleText = document.createElement('div');
            titleText.classList.add('title');
            titleText.textContent = titles[i];
            titleText.id = `title${i + 1}`;

            titlePicDiv.appendChild(titleImage);
            titlePicDiv.appendChild(titleText);

            const content2 = document.querySelector('.content2');
            content2.appendChild(titlePicDiv);
        }
    })
    .catch(error => console.error('Error:', error));
